File: dealzpark_app/lib/models/offer.dart
class Offer {
  final int id;
  final String promotionalTitle;
  final String? promotionalImageUrl;
  final int discountPercentage;
  final String? productImageUrl;
  final DateTime validFrom;
  final DateTime validTo;
  final DateTime createdAt;
  final String category;
  final int shopId;
  final String shopName;

  Offer({
    required this.id,
    required this.promotionalTitle,
    this.promotionalImageUrl,
    required this.discountPercentage,
    this.productImageUrl,
    required this.validFrom,
    required this.validTo,
    required this.createdAt,
    required this.category,
    required this.shopId,
    required this.shopName,
  });

  factory Offer.fromJson(Map<String, dynamic> json) {
    return Offer(
      id: json['id'],
      promotionalTitle: json['promotionalTitle'],
      promotionalImageUrl: json['promotionalImageUrl'],
      discountPercentage: json['discountPercentage'],
      productImageUrl: json['productImageUrl'],
      validFrom: DateTime.parse(json['validFrom']),
      validTo: DateTime.parse(json['validTo']),
      createdAt: DateTime.parse(json['createdAt']),
      category: json['category'],
      shopId: json['shopId'],
      shopName: json['shopName'],
    );
  }
}

File: DealzParkApi/Models/Offer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DealzParkApi.Models
{
    
    public class Offer
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string PromotionalTitle { get; set; }

        public string? PromotionalImageUrl { get; set; } // Optional

        [Range(0, 100)]
        public int DiscountPercentage { get; set; }

        public string? ProductImageUrl { get; set; } // Optional

        [Required]
        public DateTime ValidFrom { get; set; }

        [Required]
        public DateTime ValidTo { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        [MaxLength(100)]
        public string Category { get; set; }

        // Foreign Key for Shop
        public int ShopId { get; set; }

        // Navigation property
        [ForeignKey("ShopId")]
        public virtual Shop? Shop { get; set; }
    }
}